docs/syntax.md

```diff
@@ -60,17 +60,15 @@ const  = TSXAir(() => {
 
 ## Triggering Rendering
 
-There are 4 ways to trigger a render:
+There are 3 ways to trigger a render:
 - Changes to component properties
 - Changes to a store defined in the component
-- Changes to a [when](#using-when)/[memo](#using-memo) dependency
 - Calling `invalidate`
 
 ## "when" and "memo"
-`when` and `memo` define actions that are re-evaluated when their dependencies change.
-The dependencies can be defined or, if not specified inferred from the action code.
+`when` and `memo` allow running parts of the component logic only when a specific dependency has changed.
 ### Diffrences between when and memo:
-- a `when` action may return an "undo" function that will be called before the action is called again
+- a `when` action may return an "undo" function that will be called before the action is called again ( due to dependency change ) and when the component is about to be unmounted
```

Comment:
I think we only make a call to the **undo** function before unmounting (not the main action). I don't think an unmount is a natural change to explicit dependencies

Reply:
look at the use cases, if you subscribed you want to unsubscribe

Reply:
Yes, the **returned** ("undo") function should be executed, not the action (where you'd subscribe)